### PR TITLE
csv import attachments support and typescript migration

### LINKED ISSUE #2782: Allow admins to import attachments (images and other files) during csv import
As an administrator of an Uwazi instance that would like to migrate thousands of entities that have files attached to them (mostly images), I want to be able to migrate these files and reflect their connection to the respective entity (without having to upload these attachments manually). 

Note from Rafa: we need to keep in mind that attachments could belong to one or more languages, so this should be reflected in the import which is going to be the only tricky part.

### PR DESCRIPTION
**Pull request by daneryl**
Fixes #2782 

attachments is now supported on the csv import. for this to work the import file should be a zipFile containing an import.csv file and the files that are going to be used as attachments.

on the import.csv file, attachments can be assigned creating an "attachments" column, "|" should be used as a separator to add multiple files, like in the example below:

attachments column do NOT support "attachments__language" format only a single attachment column is supported, all attachments using import are going to be assigned to the entity in all languages.

```
Title,     attachments,
title1,    attachment1.pdf,
title2,    attachment2.pdf|attachment4.pdf, 
title3,    attachment3.pdf|attachment1.pdf, 

```

i have migrated the module to typescript and there are a lot of file changes, pls review the whole thing but in case it helps the actual implementation for attachments is here:
https://github.com/huridocs/uwazi/blob/8ce0596b042ab45247b9f821805011c4666d721c/app/api/csv/importEntity.ts#L69-L77

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

File: app/api/csv/typeParsers.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import multiselect from './typeParsers/multiselect';
+import select from './typeParsers/select';
+import relationship from './typeParsers/relationship';
+
+const defaultParser = () => async (
+  entityToImport: RawEntity,
+  property: PropertySchema
+): Promise<MetadataObjectSchema[]> => [{ value: entityToImport[ensure<string>(property.name)] }];
+
+export default {
```
Comment (konzz) [inline app/api/csv/typeParsers.ts:17]: review if the constructor for this methods is really needed

Comment (daneryl) [reply app/api/csv/typeParsers.ts:17]: removed the constructor

File: app/api/csv/specs/csvLoaderLanguages.spec.ts [hunk 2]
```diff
@@ -76,10 +77,12 @@ describe('csvLoader languages', () => {
   it('should import translated metadata properties', async () => {
     const enText = imported
       .filter(e => e.language === 'en')
-      .map(i => i.metadata.text_label[0].value);
+      // eslint-disable-next-line camelcase
```
[the remark this answers is not in the document]
Comment (daneryl) [reply app/api/csv/specs/csvLoaderLanguages.spec.ts:80]: this is based on the template property name, the import csv uses the name generation util and to be able to test this properly the property has to have a space in the label on the fixures, that are shared across all the csv tests. this generates this text_label metadata property, should i add an "textlabel" extra property on the template for this test only ?

[the remark this answers is not in the document]
Comment (konzz) [reply app/api/csv/specs/csvLoaderLanguages.spec.ts:80]: no, I think the test makes sense the way it is.

**Pull request merged**